src/pages/resultsView/ResultsViewPageStoreUtils.ts

```diff
@@ -757,6 +728,90 @@ export function parseGenericAssayGroups(
     return parsedGroups;
 }
 
+export function createDiscreteCopyNumberDataKey(
+    d: NumericGeneMolecularData | DiscreteCopyNumberData
+) {
+    return d.sampleId + '_' + d.molecularProfileId + '_' + d.entrezGeneId;
+}
+
+export function evaluateDiscreteCNAPutativeDriverInfo(
+    cnaDatum: CustomDriverNumericGeneMolecularData,
+    oncoKbDatum: IndicatorQueryResp | undefined | null | false,
+    customDriverAnnotationsActive: boolean,
+    customDriverTierSelection: ObservableMap<boolean> | undefined
+) {
+    const oncoKb = oncoKbDatum ? getOncoKbOncogenic(oncoKbDatum) : '';
+
+    // Set driverFilter to true when:
+    // (1) custom drivers active in settings menu
+    // (2) the datum has a custom driver annotation
+    const customDriverBinary: boolean =
+        (customDriverAnnotationsActive &&
+            cnaDatum.driverFilter === 'Putative_Driver') ||
+        false;
+
+    // Set tier information to the tier name when the tiers checkbox
+    // is selected for the corresponding tier of the datum in settings menu.
+    // This forces the CNA to be counted as a driver mutation.
+    const customDriverTier: string | undefined =
+        cnaDatum.driverTiersFilter &&
+        customDriverTierSelection &&
+        customDriverTierSelection.get(cnaDatum.driverTiersFilter)
+            ? cnaDatum.driverTiersFilter
+            : undefined;
+
+    return {
+        oncoKb,
+        customDriverBinary,
+        customDriverTier,
+    };
+}
+
+export function evaluateMutationPutativeDriverInfo(
+    mutation: Mutation,
+    oncoKbDatum: IndicatorQueryResp | undefined | null | false,
+    hotspotAnnotationsActive: boolean,
+    hotspotDriver: boolean,
+    cbioportalCountActive: boolean,
+    cbioportalCountExceeded: boolean,
+    cosmicCountActive: boolean,
+    cosmicCountExceeded: boolean,
+    customDriverAnnotationsActive: boolean,
+    customDriverTierSelection: ObservableMap<boolean> | undefined
+) {
+    const oncoKb = oncoKbDatum ? getOncoKbOncogenic(oncoKbDatum) : '';
+    const hotspots = hotspotAnnotationsActive && hotspotDriver;
+    const cbioportalCount = cbioportalCountActive && cosmicCountExceeded;
+    const cosmicCount = cosmicCountActive && cosmicCountExceeded;
+
+    // Set driverFilter to true when:
+    // (1) custom drivers active in settings menu
+    // (2) the datum has a custom driver annotation
+    const customDriverBinary: boolean =
```

Comment:
`!!(...)` intead of ` || false` makes it a little more concise

Reply:
Ah, updated